src/components/Tabs/TabLabel.tsx

```diff
@@ -0,0 +1,99 @@
+import styled, { css } from 'styled-components';
+
+import {
+  getColor,
+  getFontSize,
+  getLineHeight,
+  pxToRem,
+} from '../../utils/helpers';
+import { TabSizes, TabVariants } from './Tabs.enums';
+import { LabelProps, Sizes, Variants } from './Tabs.types';
+
+const largeTextSize = css`
+  font-size: ${getFontSize('lg')};
+  line-height: ${getLineHeight('lg')};
+`;
+
+const mediumTextSize = css`
+  font-size: ${getFontSize('md')};
+  line-height: ${getLineHeight('md')};
+`;
+
+const largeUnderlineSize = css`
+  font-size: ${getFontSize('h2')};
+  line-height: ${getLineHeight('xxl')};
+`;
+
+const mediumUnderlineSize = css`
+  font-size: ${getFontSize('h3')};
+  line-height: ${getLineHeight('xl')};
+`;
+
+const smallUnderlineSize = css`
+  font-size: ${getFontSize('h4')};
+  line-height: ${getLineHeight('lg')};
+`;
+
+const underlineSizes = {
+  [TabSizes.lg]: largeUnderlineSize,
+  [TabSizes.md]: mediumUnderlineSize,
+  [TabSizes.sm]: smallUnderlineSize,
+};
+
+const textSizes = {
+  [TabSizes.lg]: largeTextSize,
+  [TabSizes.md]: mediumTextSize,
+};
+
+const underlineTab = css<LabelProps & { size: Sizes; variant: Variants }>`
+  ${({ size }) => underlineSizes[size] || underlineSizes[TabSizes.md]};
+
+  padding-bottom: ${pxToRem(2)};
+  font-weight: 500;
+  border-bottom: 2px solid
+    ${({ $isSelected, $color }) =>
+      $isSelected ? getColor($color) : getColor('graphiteHB')};
+
+  &:hover,
+  &:focus {
+    color: ${getColor('graphite4B')};
+    text-decoration: none;
+    border-bottom: 2px solid ${({ $color }) => getColor($color)};
+  }
+  &:visited {
+    color: ${getColor('graphite4B')};
+  }
+`;
+
+const textTab = css<LabelProps & { size: Sizes; variant: Variants }>`
+  ${({ size }) => textSizes[size] || textSizes[TabSizes.md]};
+
+  color: ${({ $isSelected, $color }) =>
+    $isSelected ? getColor($color) : getColor('graphiteHB')};
```

Comment:
Noticed here when I try to use it that this is incorrect, should be:

default: `blueberryClassic` (I'd set this as default value of $color), selected: `graphite4B`, hover/focus: `lighten($color, xx)`